pkg/containerd/cmd/service/start.go

```diff
@@ -84,21 +84,25 @@ func start(args []string) {
 		log.WithError(err).Fatal("failed to create container")
 	}
 
-	io := func() (*containerd.IO, error) {
-		logfile := filepath.Join("/var/log", service+".log")
-		// We just need this to exist.
-		if err := ioutil.WriteFile(logfile, []byte{}, 0666); err != nil {
-			log.WithError(err).Fatal("failed to touch logfile")
-		}
-		return &containerd.IO{
-			Stdin:    "/dev/null",
-			Stdout:   logfile,
-			Stderr:   logfile,
-			Terminal: false,
-		}, nil
+	logfile := filepath.Join("/var/log", service+".log")
+	// We just need this to exist.
+	if err := ioutil.WriteFile(logfile, []byte{}, 0666); err != nil {
+		log.WithError(err).Fatal("failed to touch logfile")
+	}
+
+	// io.Reader from '/dev/null'
+	logR, err := os.Open("/dev/null")
+	if err != nil {
+		log.WithError(err).Fatal("failed to open log file")
+	}
+
+	// io.Writer for both normal and error output upon logfile
+	logOE, err := os.Open(logfile)
+	if err != nil {
+		log.WithError(err).Fatal("failed to open log file")
 	}
 
-	task, err := ctr.NewTask(ctx, io)
+	task, err := ctr.NewTask(ctx, containerd.NewIO(logR, logOE, logOE))
```

Comment:
When we do update IMHO we should just update the type of the existing function closure instead of rewriting like this. The new `id string` parameter can be ignored in this context.